sdk/search/azure-search/src/main/java/com/azure/search/SearchIndexAsyncClient.java

```diff
@@ -468,7 +465,7 @@ SearchPagedFlux search(String searchText, SearchOptions searchOptions, RequestOp
             return restClient.documents()
                 .getWithRestResponseAsync(key, selectedFields, requestOptions, context)
                 .map(res -> {
-                    Document doc = res.getValue();
+                    Document doc = new Document(res.getValue());
```

Comment:
Isn't this going to copy the entire map?

Reply:
Yes, it will copied the map.

Here is the constructor JavaDoc
```
    /**
     * Constructs a new {@code HashMap} with the same mappings as the
     * specified {@code Map}.  The {@code HashMap} is created with
     * default load factor (0.75) and an initial capacity sufficient to
     * hold the mappings in the specified {@code Map}.
     *
     * @param   m the map whose mappings are to be placed in this map
     * @throws  NullPointerException if the specified map is null
     */
    public HashMap(Map<? extends K, ? extends V> m) {
```

Reply:
That doesn't seem great for performance.

Reply:
Do we need Document to extend from HashMap, or is it sufficient that it have a map-like API?

Reply:
@JonathanGiles Strictly speaking no. I think that was done for implementation convenience. It also means that `SearchDocument` (formerly `Document`) implements `Map`. The only scenario I can imagine where that's necessary is interop with other JVM languages.